### ISSUE TITLE
[Config restore] Failure to restore pool scrub task when pool ID has changed

### ISSUE THREAD

**Issue by FroggyFlox** [rockstor/rockstor-core#2508]
This issue is identical to #2355 but concerns the scheduled tasks of type **pool scrub**.

## Reproducer
1. Create a VM with 1 OS disk + 2 data disks (Data1 and Data2).
2. Create the following pools:
    - rock-pool on Data1
    - rock-pool2 on Data2
3. Create two scheduled tasks:
    - pool scrub on rock-pool: `rockpool_scrub`
    - pool scrub on rockpool2: `rockpool2_scrub`
4. Create a config backup
5. Delete rock-pool2 and wipe the disk
6. Create a new pool named rock-pool2 using disk Data2. The pool named `rock-pool2` thus now has a different pool ID than before.
7. Restore config backup taken in step 4

Error:
```
[03/Mar/2023 15:11:40] INFO [storageadmin.views.config_backup:223] Started restoring scheduled tasks.
[03/Mar/2023 15:11:41] INFO [storageadmin.views.config_backup:57] Successfully created resource: https://localhost/api/sm/tasks. Payload: {'task_type': u'scrub', 'name': u'rockpool_scrub', 'crontabwindow': u'*-*-*-*-*-*', 'enabled': False, 'crontab': u'42 3 * * 5', 'meta': {u'pool_name': u'rock-pool', u'pool': u'2'}}
[03/Mar/2023 15:11:41] ERROR [storageadmin.views.config_backup:63] Exception occurred while creating resource: https://localhost/api/sm/tasks. Payload: {'task_type': u'scrub', 'name': u'rockpool2_scrub', 'crontabwindow': u'*-*-*-*-*-*', 'enabled': False, 'crontab': u'42 3 * * 5', 'meta': {u'pool_name': u'rock-pool2', u'pool': u'3'}}. Exception: 500 Server Error: Internal Server Error for url: https://localhost/api/sm/tasks. Moving on.
[03/Mar/2023 15:11:41] INFO [storageadmin.views.config_backup:227] Finished restoring scheduled tasks.
```

While the first scrub task was restored successfully, the second one failed:
```
ERROR [storageadmin.views.config_backup:63] Exception occurred while creating resource: https://localhost/api/sm/tasks. Payload: {'task_type': u'scrub', 'name': u'rockpool2_scrub', 'crontabwindow': u'*-*-*-*-*-*', 'enabled': False, 'crontab': u'42 3 * * 5', 'meta': {u'pool_name': u'rock-pool2', u'pool': u'3'}}. 
```

Contrary to tasks of type *snapshot*, it seems we have both the pool ID and the pool name in the json_meta. Based on `git` history, this is on purpose and for display of the pool name in UI elements (#1760):
https://github.com/rockstor/rockstor-core/blob/578c53e0422b811d096d74feb386b8c21126d247/src/rockstor/smart_manager/views/task_scheduler.py#L52-L54

The fix for this issue should be similar to #2507: validate the pool ID in the config backup to ensure it reflects the correct pool in the target system.

**Comment by phillxnet** [comment]
@FroggyFlox Nice. I did wonder about this while testing your latest pull request that is now merged.

Fancy popping in a pr with your referenced fix for this one.

As always much appreciated. I'll try and get our publishing back-end ready for our next rpm release give the recent repo changes, as I think this would be a good addition along with what I've just merged of yours.

**Comment by FroggyFlox** [comment]
Thanks @phillxnet !

I'll prepare the PR as soon as I have some time. It should (hopefully) all be ready to submit.

**Comment by FroggyFlox** [comment]
Closing as Fixed by #2513
